### ISSUE TITLE
2.0: FTS match query resulting in "Invalid LiteCore query" exception

### ISSUE THREAD

**Issue by rajagp** [couchbase/couchbase-lite-ios#1931]
--------
* Version:   DB17
* Client OS: iOS

Using match query along with other expressions in the `where` clause results in an exception
**Sample Query:** 
```
let hotelSearchQuery = Query
            .select(SelectResult.all())
            .from(DataSource.database(db))
            .where(
                Expression.property("description").match(descriptionStr)
                    .and(Expression.property("type").equalTo("hotel"))
                    .and(Expression.property("country").equalTo(locationStr)
                        .or(Expression.property("city").equalTo(locationStr))
                        .or(Expression.property("state").equalTo(locationStr))
                        .or(Expression.property("address").equalTo(locationStr))))
```
CouchbaseLite Query Info: Query encoded as 
```
{
  "WHERE": [
    "AND",
    [
      "AND",
      [
        "MATCH",
        [
          ".description"
        ],
        "%Optional(\"Pets allowed\")%"
      ],
      [
        "=",
        [
          ".type"
        ],
        "hotel"
      ]
    ],
    [
      "OR",
      [
        "OR",
        [
          "OR",
          [
            "=",
            [
              ".country"
            ],
            "London"
          ],
          [
            "=",
            [
              ".city"
            ],
            "London"
          ]
        ],
        [
          "=",
          [
            ".state"
          ],
          "London"
        ]
      ],
      [
        "=",
        [
          ".address"
        ],
        "London"
      ]
    ]
  ],
  "WHAT": [
    [
      "."
    ]
  ]
}

 ```
**Resulting Exception:**
CouchbaseLite  WARNING: Invalid LiteCore query: MATCH can only appear at top-level, or in a top-level AND

**Expected:**
Since the `match` is at top level ANS expression, expect this to work.

(CC @snej )

**Comment by djpongh** [comment]
Fixed in LiteCore for all platforms.

**Comment by snej** [comment]
Fixed in LiteCore commit 289f256.
